### PR TITLE
cgifsave: theshold alpha and check alpha constraint in one pass

### PR DESCRIPTION
**Pull request by DarthSim**
Hey!

I noticed that we can threshold alpha and check for alpha constraint in a single pass. This doesn't increase speed dramatically yet saves some milliseconds.

### CODE REVIEW

File: libvips/foreign/cgifsave.c [hunk 1]
```diff
@@ -234,21 +215,32 @@ vips_foreign_save_cgif_write_frame( VipsForeignSaveCgif *cgif )
...
-			p[0] = 0;
-			p[1] = 0;
-			p[2] = 0;
-			p[3] = 0;
+			cur[0] = 0;
+			cur[1] = 0;
+			cur[2] = 0;
+			cur[3] = 0;
+
+			if( bef && bef[i*4+3] != 0 )
```
[the remark this answers is not in the document]
Comment (jcupitt) [reply libvips/foreign/cgifsave.c:239]: Yes, exactly. Just a silly layout thing.

[the remark this answers is not in the document]
Comment (DarthSim) [reply libvips/foreign/cgifsave.c]: Got it. I mostly write in Go, and we remove spaces between square brackets there, so I'm kinda used to this :)

File: libvips/foreign/cgifsave.c [hunk 2]
```diff
@@ -183,26 +183,6 @@ vips_foreign_save_cgif_pixels_are_equal( const VipsPel *cur, const VipsPel *bef,
 	return( sqrt( dR * dR + dG * dG + dB * dB ) <= maxerror );
```
[the remark this answers is not in the document]
Comment (DarthSim) [reply libvips/foreign/cgifsave.c]: Thought about this myself but decided not to fix this here as this seemed not directly related to this PR. Fixed in a separate commit.

File: libvips/foreign/cgifsave.c [hunk 3]
```diff
@@ -217,8 +197,9 @@ vips_foreign_save_cgif_write_frame( VipsForeignSaveCgif *cgif )
 	VipsPel *frame_bytes = 
 		VIPS_REGION_ADDR( cgif->frame, 0, frame_rect->top );
 
-	VipsPel * restrict p;
 	int i;
+	VipsPel *cur, *bef;
```
[the remark this answers is not in the document]
Comment (DarthSim) [reply libvips/foreign/cgifsave.c]: Fixed

**Comment by jcupitt** [comment]
Thanks!

**Pull request merged**